internal/vcs/git.go

```diff
@@ -96,6 +96,9 @@ func NewGitRepo(ctx context.Context, storagePath string, opts GitRepoOptions) (*
 
 	checkoutErr := checkout(opts.Revision, repo)
 	if checkoutErr != nil {
+		if errors.Is(checkoutErr, plumbing.ErrReferenceNotFound) {
+			return nil, InvalidRevisionError{opts.Revision}
+		}
 		return nil, UpdateFailedError{
```

Comment:
just for consistency, could we return checkoutErr instead of UpdateFailedError?
If this code returns nil, UpdateFailedError at this stage, it would panic like it did in the ticket. I don't think that it's possible right now that it happens but it makes the code more robust